src/backend/access/transam/varsup.c

```diff
@@ -514,6 +520,76 @@ ForceTransactionIdLimitUpdate(void)
 	return false;
 }
 
+Oid
+GetNewTempObjectId(void)
+{
+	Oid			result;
+	Oid			tempOidStart;
+	static Oid 	nextTempOid = InvalidOid;
+
+	/* safety check, we should never get this far in a HS standby */
+	if (RecoveryInProgress())
```

Comment:
A test case for this would require setting up a hot standby server. And as in the comment, we shouldn't reach this codepath in that case as well. I added notes in the quip document.